app/main.py

```diff
@@ -1 +1,24 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import NotVaccinatedError
+from app.errors import OutdatedVaccineError
+from app.errors import NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotVaccinatedError:
+            return "All friends should be vaccinated"
+        except OutdatedVaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_to_buy += 1
+        except KeyError:
+            return "All friends should be vaccinated"
```

Comment:
Catching `KeyError` here might be misleading since it is not clear what key is missing. Consider handling this case more explicitly in the `visit_cafe` method of the `Cafe` class, or provide a more descriptive error message.